pyomeca/analogs.py

```diff
@@ -1,371 +1,508 @@
-                # get index where assignment are empty
-                nan_idx = [i for i, v in enumerate(iassign) if not v]
-                if nan_idx:
-                    iassign_without_nans = [i for i in iassign if i]
-                else:
-                    iassign_without_nans = iassign
-
-                try:
-                    emg = Analogs3d.from_c3d(
-                        f"{itrial}", prefix=":", names=iassign_without_nans
-                    )
-                    if nan_idx:
-                        # if there is any empty assignment, fill the dimension with nan
-                        emg.get_nan_idx = np.array(nan_idx)
-                        for i in nan_idx:
-                            emg = np.insert(emg, i, np.nan, axis=1)
-                        # check if nan dimension are correctly inserted
-                        n = np.isnan(emg).sum(axis=2).ravel()
-                        if not np.array_equal(n.argsort()[-len(nan_idx) :], nan_idx):
-                            raise ValueError("NaN dimensions misplaced")
-                        print(f"\t{itrial.stem} (NaNs: {nan_idx})")
-                    else:
-                        print(f"\t{itrial.stem}")
-
-                    # check if dimensions are ok
-                    if not emg.shape[1] == len(iassign):
-                        raise ValueError("Wrong dimensions")
-                    break
-                except IndexError:
-                    emg = []
-
-            if np.any(emg):
-                trials.append(emg)
-            else:
-                raise ValueError(
-                    f"no assignments were found for the trial {itrial.stem}"
-                )
+        Analogs DataArray from a mot file.
 
-        return trials
+        Arguments:
+            filename: Any valid string path
+            end_header: Index where `endheader` appears (0 indexed). If not provided, the index is automatically determined.
+            kwargs: Keyword arguments to be passed to `from_csv`
 
-    def process_trials(self):
-        """Process trials from a list and concatenate them in a single dict"""
-        print("Processing trials...")
-        concatenated = {
-            imuscle: np.array([]) for imuscle in range(self.trials[0].shape[1])
-        }
+        Returns:
+            Analogs `xarray.DataArray` with the specified data and coordinates
 
-        for i, itrial in enumerate(self.trials):
-            # emg processing
-            itrial = (
-                itrial.band_pass(
-                    freq=itrial.get_rate, order=self.order, cutoff=self.band_pass_cutoff
-                )
-                .center()
-                .rectify()
-                .low_pass(
-                    freq=itrial.get_rate, order=self.order, cutoff=self.low_pass_cutoff
-                )
-            )
+        !!! example
+            To read [this mot file](https://github.com/pyomeca/pyomeca/blob/master/tests/data/inverse_kin.mot),
```

Comment:
No because the docstring is used to generated the API reference in the docmentation website.
if I turn it into a relative path, the file won't be reachable on the internet.